src/uct/ib/dc/dc_mlx5.c

```diff
@@ -440,6 +442,14 @@ ucs_status_t uct_dc_mlx5_iface_dci_connect(uct_dc_mlx5_iface_t *iface,
         return UCS_ERR_IO_ERROR;
     }
 
+    dev = uct_ib_iface_device(&iface->super.super.super);
+    memset(&ibv_ece, 0, sizeof(ibv_ece));
+    if (dev->flags & UCT_IB_DEVICE_FLAG_ECE) {
+        uct_ib_query_ece(dci->txwq.super.verbs.qp, &ibv_ece);
+        ibv_ece.options = iface->super.super.config.conn_ece;
+        uct_ib_set_ece(dci->txwq.super.verbs.qp, &ibv_ece);
+    }
```

Comment:
As been explained, ```config.ece``` will be ```0xffffffffU``` or ```0x0``` under using ```auto``` configuration. It can use the ```config.ece``` value here directly.